autocorpus/reference.py

```diff
@@ -1,11 +1,33 @@
 """Use regular expression for searching/replacing reference strings."""
 
 import re
+from dataclasses import dataclass
 from typing import Any
 
+from .data_structures import Paragraph
 
-def get_references(reference: dict[str, Any], section_heading: str) -> dict[str, Any]:
-    """Retrieve a structured reference dictionary from a BeautifulSoup object and section heading.
+
+@dataclass
+class ReferencesParagraph(Paragraph):
+    """A paragraph for the references section of the article."""
+
+    title: str = ""
+    journal: str = ""
+    volume: str = ""
+
+    def as_dict(self) -> dict[str, Any]:
+        """Return the dictionary representation of the ReferencesParagraph."""
+        return {
+            k: v
+            for k, v in super().as_dict().items()
+            if v or k not in ("title", "journal", "volume")
```

Comment:
Python will interpret this as:

```py
            if v or (k not in ("title", "journal", "volume"))
```

So this will always be true if `v` is non-empty.

This will work, but might not be v readable:

```suggestion
            if not {k, v}.isdisjoint({"title", "journal", "volume"})
```

Reply:
I've just realised that you probably meant it to work that way :laughing:. I'll leave my comment there for the lols

Anyway, maybe add a comment to say what it's supposed to do.